apps/project/tests/test_schemas.py

```diff
@@ -865,6 +870,134 @@ def _expected_response(
+
+        project_with_access = [projects[0], projects[2]]
+
+        for project in project_with_access:
+            project.add_member(user, role=self.project_role_member)
+
+        def _query_check(project_id):
+            return self.query_check(QUERY, variables={'projectId': project_id})
+
+        self.force_login(user)
+
+        with patch('deep.trackers.timezone.now') as timezone_now_mock:
+            timezone_now_mock.return_value = timezone_now = datetime(2021, 1, 1, 0, 0, 0, 123456, tzinfo=pytz.UTC)
+            # Normal run without calling notifications endpoint
+            for project in projects:
+                assert project.last_read_access is None
+                _query_check(project.id)['data']['project']
+                with self.captureOnCommitCallbacks(execute=True):
+                    schedule_tracker_data_handler()
+                if project in project_with_access:
+                    project.refresh_from_db()
+                    assert project.last_read_access == timezone_now
+                else:
+                    assert project.last_read_access is None
+
+            timezone_now_mock.return_value = new_timezone_now = datetime(2022, 1, 1, 0, 0, 0, 123456, tzinfo=pytz.UTC)
+            # Normal run without calling notifications endpoint
+            for project in projects:
+                if project in project_with_access:
+                    assert project.last_read_access == timezone_now
+                else:
+                    assert project.last_read_access is None
+                _query_check(project.id)['data']['project']
+                with self.captureOnCommitCallbacks(execute=True):
+                    schedule_tracker_data_handler()
+                if project in project_with_access:
+                    project.refresh_from_db()
+                    assert project.last_read_access == new_timezone_now
+                else:
+                    assert project.last_read_access is None
+
+    def test_project_last_write_access(self):
+        MUTATION = '''
+            mutation MyMutation ($projectId: ID!) {
+              project(id: $projectId) {
+                  id
+                }
+            }
+        '''
+
+        user = UserFactory.create()
+        projects = ProjectFactory.create_batch(2)
+        projects.extend(
+            ProjectFactory.create_batch(2, is_private=True)
+        )
+
+        project_with_access = [projects[0], projects[2]]
+
+        for project in project_with_access:
+            project.add_member(user, role=self.project_role_member)
+
+        def _query_check(project_id, **kwargs):
+            return self.query_check(MUTATION, variables={'projectId': project_id}, **kwargs)
+
+        self.force_login(user)
+
+        with patch('deep.trackers.timezone.now') as timezone_now_mock:
+            timezone_now_mock.return_value = timezone_now = datetime(2021, 1, 1, 0, 0, 0, 123456, tzinfo=pytz.UTC)
```

Comment:
Please add notes on what this test case is trying to achieve.

Reply:
Done https://github.com/the-deep/server/commit/3a19f3fd39d0d5d09253f94d32acdb1c4db0f953